rfd/0002/README.mdx

```diff
@@ -0,0 +1,198 @@
+
+Random idea: we could have those be state machines. In state machine mode we would not have concurrency enabled. in standard mode they could be scaled horizontally.
+
+### Table Writer Process
+
+This is a generic piece of code that reads anything that is written back from the ingest function to the streaming function and writes it to the corresponding table.
+
+We could be using [vector.dev](https://vector.dev/) for this
+
+### CLI
+
+The igloo CLI is a tool to help the user get setup really quickly. It helps with
+
+- Initial project setup
+- Runs and configures locally the OLAP DB and the streaming engine
+- Orchestrate the ingest functions and runs them
+- Orchestrate the Table Writer
+- Runs the API server locally
+- Runs the developper console for the project
+
+## Ingest MVP Definition
+
+We are currently on getting an MVP of the Ingest flow in order to start dog fooding the project. Ie we want to start instrumenting igloo with igloo for usage metrics.
+
+All the boxes in yellow below are part of the Ingest MVP
+
+In that MVP we need the data:
+
+- To be emitted from the instrumented app
+- To Go through the web server
+- Through the streaming engine
+- Be written in a table on the OLAP DB
+- Be Queriable in the OLAP DB through its client
+
+A User needs to be able to:
+
+- Iterate on the schema definition of Ingestion tables
+- Create new Ingest Tables and Events
+- Query the data that landed in the table
+- Deploy and Iterate on the Production deployment
+
+```mermaid
+---
+title: Overall Architecture Diagram
+---
+flowchart LR
+  subgraph Igloo Server
+    HTTP[HTTP Server] --Pushes--> Stream[(Streaming Engine)]
+    LambdaI[Lambda Ingestion Function] <-- Pulls --> Stream
+    LambdaI --Pushes--> Stream
+    Table[Table Writer Process] <--Pulls--> Stream
+    Table --Writes-->OLAPI[(Ingest Table)]
+    View[(Materialized View)] --Reads--> OLAPI
+    HTTP --Queries-->View
+  end
+  subgraph User Apps
+    sdk1[Data Ingest SDK]--Pushes-->HTTP
+    sdk2[Data Consumption SDK]--Queries--> HTTP
+    sdk3[Data Vizualization SDK]--Uses-->sdk2
+  end
+
+  style sdk1 fill:#E7D182,color:#000
+  style Stream fill:#E7D182,color:#000
+  style Table fill:#E7D182,color:#000
+  style OLAPI fill:#E7D182,color:#000
+  style View fill:#E7D182,color:#000
+  style HTTP fill:#E7D182,color:#000
+```
+
+### Open Questions
+
+- In Production how to we deploy everything?
+- Do we provide configurations for the OLAP cluster?
+- Do we provide configurations for the streaming cluster?
+- Do we point the user to cloud offerings for storage to be plugged in?
+- How scalable do we need the first version of this to be?
+- Should we just bundle this all in one docker container?
+
+# Appendix
+
+## Technologies we chose
+
+### Rust
+
+As a high performant language we chose rust to be the glue we write that can be common across multiple user environments. Whether the user is leveraging a higher level language like Typescript or Javascript.
```

Comment:
what was the reason behind the choosing Rust ?

adding the why behind this choice of using Rust  could help document valuable context.